pkg/osquery/runtime/runtime_posix_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build !windows
+// +build !windows
+
+package runtime
+
+import (
+	"fmt"
+	"log/slog"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	typesMocks "github.com/kolide/launcher/ee/agent/types/mocks"
+	"github.com/kolide/launcher/pkg/log/multislogger"
+	"github.com/kolide/launcher/pkg/threadsafebuffer"
+	"github.com/osquery/osquery-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func requirePgidMatch(t *testing.T, pid int) {
+	pgid, err := syscall.Getpgid(pid)
+	require.NoError(t, err)
+	require.Equal(t, pgid, pid)
+}
+
+// TestOsquerySlowStart tests that the launcher can handle a slow-starting osqueryd process.
+// This this is only enabled on non-Windows platforms because suspending we have not yet
+// figured out how to suspend a process on windows via golang.
```

Comment:
(nit) grammar seems wrong. I think I know what you mean, but I'm not totally sure

Reply:
updated